ngramoffset.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zoekt
+
+import (
+	"sort"
+)
+
+// simple interface type check for implementations
+var (
+	_ ngramOffset = &arrayNgramOffset{}
+)
+
+type ngramOffset interface {
```

Comment:
do we still need the interface, or can we just rely on the concrete type `arrayNgramOffset`? Additionally the zero value for `arrayNgramOffset` behaves nicely, which is a nice property to have to avoid needing to initialize it for indexData.

Reply:
done.